registry/registry_test.go

```diff
@@ -0,0 +1,54 @@
+package registry
+
+import (
+	"testing"
+
+	"github.com/OrlandoRomo/academy-go-q32021/infrastructure/api"
+	"github.com/OrlandoRomo/academy-go-q32021/interface/controller"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	testApiKey = "cc2464e4a1mshb5ceeca91e5a6adp1fa80bjsn4b48e2408b87"
+)
+
+func TestNewRegistry(t *testing.T) {
+	testcases := []struct {
+		name        string
+		message     string
+		urbanClient *api.UrbanDictionary
+	}{
+		{
+			name:        "success - urban client set up",
+			message:     "urban client in registry set up",
+			urbanClient: api.NewUrbanDictionary(testApiKey),
+		},
+		{
+			name:        "failure - urban client nil",
+			message:     "registry app with a nil urban client",
+			urbanClient: nil,
+		},
+	}
+
+	for _, test := range testcases {
+		t.Run(test.name, func(t *testing.T) {
+			if test.urbanClient == nil {
```

Comment:
It seems like this if statements are redundant, why double checking if something is nil or not? Assert already takes care of that, it would be better to add a field to the tests to indicate if you expect a nil or not, and do something like `assert.Equal(t, test.expectedClient, test.urbanClient)`

And call the NewUrbanDictionary function inside this test body